database/errors.go

```diff
@@ -32,3 +33,13 @@ func IsErrConstraintUnique(err error) bool {
 
 	return false
 }
+
+// IsErrNotFound returns true if the input error was returned by SQLite due
+// to a missing record.
+func IsErrNotFound(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	return errors.Is(err, sql.ErrNoRows)
```

Comment:
Just verify that we don't need to check for Dqlite error types, as in https://github.com/juju/juju/pull/15547.

Reply:
There was nothing from [https://github.com/mattn/go-sqlite3/blob/master/error.go#L37](here) that jumped out.